includes/bylines/class-bylines.php

```diff
@@ -294,5 +298,69 @@ public static function newspack_blocks_post_byline( $byline ) {
 
 		return $custom_byline;
 	}
+
+	/**
+	 * Filters the post meta data for distribution and add a new method with a mapping of author IDs to author emails.
+	 *
+	 * @param array   $meta The post meta data.
+	 * @param WP_Post $post The post object.
+	 */
+	public static function newspack_network_distributed_post_meta( $meta, $post ) {
+		$byline_is_active = \get_post_meta( \get_the_ID(), self::META_KEY_ACTIVE, true );
+		if ( ! $byline_is_active ) {
+			return $meta;
+		}
+
+		$byline = \get_post_meta( $post->ID, self::META_KEY_BYLINE, true );
+
+		$author_ids = self::extract_author_ids_from_shortcode( $byline );
+
+		if ( empty( $author_ids ) ) {
+			return $meta;
+		}
+
+		$mapping = [];
+		// create a mapping of author IDs to author emails.
+		foreach ( $author_ids as $author_id ) {
+			$mapping[ $author_id ] = get_the_author_meta( 'user_email', $author_id );
+		}
+
+		$meta['_newspack_byline_network_authors'] = [ wp_json_encode( $mapping ) ];
+
+		return $meta;
+	}
+
+	/**
+	 * After an incoming post is inserted, update the IDs with the IDs in the target site, based on the mapping that was sent.
+	 *
+	 * @param int   $post_id   The post ID.
+	 * @param bool  $is_linked Whether the post is linked.
+	 * @param array $payload   The post payload.
+	 */
+	public static function newspack_network_incoming_post_inserted( $post_id, $is_linked, $payload ) {
+		if ( ! $is_linked ) {
+			return;
+		}
+
+		$byline = \get_post_meta( $post_id, self::META_KEY_BYLINE, true );
```

Comment:
Small nit: The mixed use of backslashed WP functions is a bit confusing. I don't have a preference for either, but I do like it consistent.